src/mappings/primitives.ts

```diff
@@ -105,3 +128,20 @@ export async function handleEvent(event: CosmosEvent): Promise<void> {
 
   await eventEntity.save();
 }
+
+async function _handleBlockError(err: Error, _: CosmosBlock): Promise<void> {
+  // NB: we won't have persisted any related entities yet.
+  await trackUnprocessed(err, {});
+}
+
+async function _handleTransactionError(err: Error, tx: CosmosTransaction): Promise<void> {
+  await trackUnprocessed(err, primitivesFromTx(tx));
+}
+
+async function _handleMessageError(err: Error, msg: CosmosMessage): Promise<void> {
+  await trackUnprocessed(err, primitivesFromMsg(msg));
+}
+
+async function _handleEventError(err: Error, event: CosmosEvent): Promise<void> {
```

Comment:
This is redundant and its usage can be replaced with `unprocessedEventHandler` from utils.